lib/IRGen/IRGen.cpp

```diff
@@ -601,10 +602,29 @@ bool swift::compileAndWriteLLVM(llvm::Module *module,
     EmitPasses.add(createPrintModulePass(out));
     break;
   case IRGenOutputKind::LLVMBitcode: {
+    // Emit a module summary by default for Regular LTO except ld64-based ones
+    // (which use the legacy LTO API).
+    bool EmitRegularLTOSummary =
+        targetMachine->getTargetTriple().getVendor() != llvm::Triple::Apple;
```

Comment:
Why the vendor check rather than `!targetMachine->getTargetTriple().isOSDarwin()`?

Reply:
@kubamracek Which form do we prefer here?  `!= Apple` or `isOSDarwin`?

Reply:
I followed what clang does.
https://github.com/llvm/llvm-project/blob/a5040860412f3e61151e427af1cff6ff0aebb775/clang/lib/CodeGen/BackendUtil.cpp#L164-L172